File: media-attachments/javascript/media-attachments.js
/**
 * SendSeven API - Media Attachments Example (JavaScript)
 *
 * Demonstrates how to upload files and send media messages using the SendSeven API.
 */

require('dotenv').config();
const fs = require('fs');
const path = require('path');

// Configuration from environment
const API_TOKEN = process.env.SENDSEVEN_API_TOKEN;
const TENANT_ID = process.env.SENDSEVEN_TENANT_ID;
const API_URL = process.env.SENDSEVEN_API_URL || 'https://api.sendseven.com/api/v1';
const CONVERSATION_ID = process.env.CONVERSATION_ID;

// File size limits (in bytes)
const FILE_SIZE_LIMITS = {
  image: 16 * 1024 * 1024,     // 16 MB
  document: 100 * 1024 * 1024, // 100 MB
  video: 16 * 1024 * 1024,     // 16 MB
  audio: 16 * 1024 * 1024,     // 16 MB
};

// Supported content types by message type
const SUPPORTED_TYPES = {
  image: ['image/jpeg', 'image/png', 'image/gif', 'image/webp'],
  document: [
    'application/pdf',
    'application/msword',
    'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
    'application/vnd.ms-excel',
    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
    'application/vnd.ms-powerpoint',
    'application/vnd.openxmlformats-officedocument.presentationml.presentation',
    'text/plain',
  ],
  video: ['video/mp4', 'video/3gpp'],
  audio: ['audio/aac', 'audio/mpeg', 'audio/ogg', 'audio/amr', 'audio/opus'],
};

// Extension to content type mapping
const CONTENT_TYPES = {
  // Images
  '.jpg': 'image/jpeg',
  '.jpeg': 'image/jpeg',
  '.png': 'image/png',
  '.gif': 'image/gif',
  '.webp': 'image/webp',
  // Documents
  '.pdf': 'application/pdf',
  '.doc': 'application/msword',
  '.docx': 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
  '.xls': 'application/vnd.ms-excel',
  '.xlsx': 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
  '.ppt': 'application/vnd.ms-powerpoint',
  '.pptx': 'application/vnd.openxmlformats-officedocument.presentationml.presentation',
  '.txt': 'text/plain',
  // Video
  '.mp4': 'video/mp4',
  '.3gp': 'video/3gpp',
  // Audio
  '.aac': 'audio/aac',
  '.mp3': 'audio/mpeg',
  '.ogg': 'audio/ogg',
  '.amr': 'audio/amr',
  '.opus': 'audio/opus',
};

/**
 * Get content type from file extension.
 * @param {string} filePath - Path to the file
 * @returns {string} The content type
 */
function getContentType(filePath) {
  const ext = path.extname(filePath).toLowerCase();
  return CONTENT_TYPES[ext] || 'application/octet-stream';
}

/**
 * Get message type from content type.
 * @param {string} contentType - The content type
 * @returns {string} The message type (image, document, video, audio)
 */
function getMessageType(contentType) {
  for (const [msgType, types] of Object.entries(SUPPORTED_TYPES)) {
    if (types.includes(contentType)) {
      return msgType;
    }
  }
  throw new Error(`Unsupported content type: ${contentType}`);
}

/**
 * Upload a file as an attachment.
 * @param {string} filePath - Path to the file to upload
 * @returns {Promise<Object>} The created attachment object
 */
async function uploadAttachment(filePath) {
  // Validate file exists
  if (!fs.existsSync(filePath)) {
    throw new Error(`File not found: ${filePath}`);
  }

  const fileStats = fs.statSync(filePath);
  const filename = path.basename(filePath);
  const contentType = getContentType(filePath);
  const messageType = getMessageType(contentType);

  // Check file size
  const maxSize = FILE_SIZE_LIMITS[messageType];
  if (fileStats.size > maxSize) {
    throw new Error(`File too large: ${fileStats.size} bytes (max ${maxSize} bytes for ${messageType})`);
  }

  // Read file content
  const fileContent = fs.readFileSync(filePath);

  // Create form data boundary
  const boundary = '----FormBoundary' + Math.random().toString(36).substring(2);

  // Build multipart form data manually
  const body = Buffer.concat([
    Buffer.from(`--${boundary}\r\n`),
    Buffer.from(`Content-Disposition: form-data; name="file"; filename="${filename}"\r\n`),
    Buffer.from(`Content-Type: ${contentType}\r\n\r\n`),
    fileContent,
    Buffer.from(`\r\n--${boundary}--\r\n`),
  ]);

  const response = await fetch(`${API_URL}/attachments`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_TOKEN}`,
      'X-Tenant-ID': TENANT_ID,
      'Content-Type': `multipart/form-data; boundary=${boundary}`,
    },
    body: body,
  });

  if (response.status === 413) {
    throw new Error('File too large (server rejected)');
  } else if (response.status === 415) {
    throw new Error('Unsupported media type (server rejected)');
  }

  if (!response.ok) {
    const errorBody = await response.text();
    throw new Error(`API Error ${response.status}: ${errorBody}`);
  }

  return response.json();
}

/**
 * Send a message with an attachment.
 * @param {string} conversationId - The UUID of the conversation
 * @param {string} attachmentId - The UUID of the uploaded attachment
 * @param {string} messageType - Type of message (image, document, video, audio)
 * @param {string} [caption] - Optional text caption for the message
 * @returns {Promise<Object>} The created message object
 */
async function sendMediaMessage(conversationId, attachmentId, messageType, caption = null) {
  const payload = {
    conversation_id: conversationId,
    message_type: messageType,
    attachments: [attachmentId],
  };

  if (caption) {
    payload.text = caption;
  }

  const response = await fetch(`${API_URL}/messages`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_TOKEN}`,
      'X-Tenant-ID': TENANT_ID,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload),
  });

  if (!response.ok) {
    const errorBody = await response.text();
    throw new Error(`API Error ${response.status}: ${errorBody}`);
  }

  return response.json();
}

/**
 * Download an attachment by ID.
 * @param {string} attachmentId - The UUID of the attachment
 * @param {string} outputPath - Path to save the downloaded file
 * @returns {Promise<void>}
 */
async function downloadAttachment(attachmentId, outputPath) {
  const response = await fetch(`${API_URL}/attachments/${attachmentId}/download`, {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${API_TOKEN}`,
      'X-Tenant-ID': TENANT_ID,
    },
  });

  if (!response.ok) {
    const errorBody = await response.text();
    throw new Error(`API Error ${response.status}: ${errorBody}`);
  }

  const buffer = await response.arrayBuffer();
  fs.writeFileSync(outputPath, Buffer.from(buffer));
  console.log(`Downloaded to: ${outputPath}`);
}

/**
 * Upload and send an image message.
 * @param {string} conversationId - The UUID of the conversation
 * @param {string} filePath - Path to the image file
 * @param {string} [caption] - Optional caption for the image
 * @returns {Promise<Object>} The created message object
 */
async function sendImage(conversationId, filePath, caption = null) {
  console.log(`Uploading image: ${filePath}`);
  const attachment = await uploadAttachment(filePath);
  console.log(`  Uploaded: ${attachment.id}`);

  console.log('Sending image message...');
  return sendMediaMessage(conversationId, attachment.id, 'image', caption);
}

/**
 * Upload and send a document message.
 * @param {string} conversationId - The UUID of the conversation
 * @param {string} filePath - Path to the document file
 * @param {string} [caption] - Optional caption for the document
 * @returns {Promise<Object>} The created message object
 */
async function sendDocument(conversationId, filePath, caption = null) {
  console.log(`Uploading document: ${filePath}`);
  const attachment = await uploadAttachment(filePath);
  console.log(`  Uploaded: ${attachment.id}`);

  console.log('Sending document message...');
  return sendMediaMessage(conversationId, attachment.id, 'document', caption);
}

/**
 * Upload and send a video message.
 * @param {string} conversationId - The UUID of the conversation
 * @param {string} filePath - Path to the video file
 * @param {string} [caption] - Optional caption for the video
 * @returns {Promise<Object>} The created message object
 */
async function sendVideo(conversationId, filePath, caption = null) {
  console.log(`Uploading video: ${filePath}`);
  const attachment = await uploadAttachment(filePath);
  console.log(`  Uploaded: ${attachment.id}`);

  console.log('Sending video message...');
  return sendMediaMessage(conversationId, attachment.id, 'video', caption);
}

/**
 * Upload and send an audio message.
 * @param {string} conversationId - The UUID of the conversation
 * @param {string} filePath - Path to the audio file
 * @param {string} [caption] - Optional caption for the audio
 * @returns {Promise<Object>} The created message object
 */
async function sendAudio(conversationId, filePath, caption = null) {
  console.log(`Uploading audio: ${filePath}`);
  const attachment = await uploadAttachment(filePath);
  console.log(`  Uploaded: ${attachment.id}`);

  console.log('Sending audio message...');
  return sendMediaMessage(conversationId, attachment.id, 'audio', caption);
}

/**
 * Demo: Upload a file and send it as a message.
 * Automatically detects the appropriate message type.
 */
async function demoUploadAndSend(filePath) {
  const contentType = getContentType(filePath);
  const messageType = getMessageType(contentType);

  console.log(`\n--- Sending ${messageType} ---`);
  console.log(`File: ${filePath}`);
  console.log(`Content-Type: ${contentType}`);

  const attachment = await uploadAttachment(filePath);
  console.log('Attachment uploaded:');
  console.log(`  ID: ${attachment.id}`);
  console.log(`  Filename: ${attachment.filename}`);
  console.log(`  Size: ${attachment.file_size} bytes`);

  const message = await sendMediaMessage(
    CONVERSATION_ID,
    attachment.id,
    messageType,
    `Here's a ${messageType} file!`
  );

  console.log('Message sent:');
  console.log(`  ID: ${message.id}`);
  console.log(`  Status: ${message.status}`);
  console.log(`  Created at: ${message.created_at}`);

  return message;
}

async function main() {
  // Validate configuration
  if (!API_TOKEN) {
    console.error('Error: SENDSEVEN_API_TOKEN environment variable is required');
    process.exit(1);
  }

  if (!TENANT_ID) {
    console.error('Error: SENDSEVEN_TENANT_ID environment variable is required');
    process.exit(1);
  }

  if (!CONVERSATION_ID) {
    console.error('Error: CONVERSATION_ID environment variable is required');
    process.exit(1);
  }

  console.log('SendSeven Media Attachments Example');
  console.log('='.repeat(40));
  console.log(`API URL: ${API_URL}`);
  console.log(`Conversation: ${CONVERSATION_ID}`);

  // Check for command line argument (file to upload)
  const filePath = process.argv[2];

  if (filePath) {
    try {
      await demoUploadAndSend(filePath);
    } catch (error) {
      console.error(`Error: ${error.message}`);
      process.exit(1);
    }
  } else {
    console.log('\nUsage: node media-attachments.js <file_path>');
    console.log('\nSupported file types:');
    console.log('  Images:    .jpg, .jpeg, .png, .gif, .webp (max 16 MB)');
    console.log('  Documents: .pdf, .doc, .docx, .xls, .xlsx, .ppt, .pptx, .txt (max 100 MB)');
    console.log('  Video:     .mp4, .3gp (max 16 MB)');
    console.log('  Audio:     .aac, .mp3, .ogg, .amr, .opus (max 16 MB)');
    console.log('\nExample:');
    console.log('  node media-attachments.js /path/to/image.jpg');

    // Demo with a sample file if it exists
    const sampleFiles = ['sample.jpg', 'sample.png', 'sample.pdf'];
    for (const sample of sampleFiles) {
      if (fs.existsSync(sample)) {
        console.log(`\nFound sample file: ${sample}`);
        try {
          await demoUploadAndSend(sample);
        } catch (error) {
          console.error(`Error: ${error.message}`);
        }
        break;
      }
    }
  }
}

// Export functions for use as a module
module.exports = {
  uploadAttachment,
  sendMediaMessage,
  downloadAttachment,
  sendImage,
  sendDocument,
  sendVideo,
  sendAudio,
  getContentType,
  getMessageType,
};

main();

File: media-attachments/java/MediaAttachments.java
/**
 * SendSeven API - Media Attachments Example (Java)
 *
 * Demonstrates how to upload files and send media messages using the SendSeven API.
 */

import java.io.*;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.*;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ArrayNode;
import com.fasterxml.jackson.databind.node.ObjectNode;

public class MediaAttachments {

    private final String apiToken;
    private final String tenantId;
    private final String apiUrl;
    private final String conversationId;
    private final HttpClient httpClient;
    private final ObjectMapper objectMapper;

    // File size limits (in bytes)
    private static final long IMAGE_MAX_SIZE = 16L * 1024 * 1024;      // 16 MB
    private static final long DOCUMENT_MAX_SIZE = 100L * 1024 * 1024;  // 100 MB
    private static final long VIDEO_MAX_SIZE = 16L * 1024 * 1024;      // 16 MB
    private static final long AUDIO_MAX_SIZE = 16L * 1024 * 1024;      // 16 MB

    // Supported content types by message type
    private static final Map<String, List<String>> SUPPORTED_TYPES = new HashMap<>();
    static {
        SUPPORTED_TYPES.put("image", Arrays.asList("image/jpeg", "image/png", "image/gif", "image/webp"));
        SUPPORTED_TYPES.put("document", Arrays.asList(
            "application/pdf", "application/msword",
            "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
            "application/vnd.ms-excel",
            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
            "application/vnd.ms-powerpoint",
            "application/vnd.openxmlformats-officedocument.presentationml.presentation",
            "text/plain"
        ));
        SUPPORTED_TYPES.put("video", Arrays.asList("video/mp4", "video/3gpp"));
        SUPPORTED_TYPES.put("audio", Arrays.asList("audio/aac", "audio/mpeg", "audio/ogg", "audio/amr", "audio/opus"));
    }

    // Extension to content type mapping
    private static final Map<String, String> CONTENT_TYPES = new HashMap<>();
    static {
        // Images
        CONTENT_TYPES.put(".jpg", "image/jpeg");
        CONTENT_TYPES.put(".jpeg", "image/jpeg");
        CONTENT_TYPES.put(".png", "image/png");
        CONTENT_TYPES.put(".gif", "image/gif");
        CONTENT_TYPES.put(".webp", "image/webp");
        // Documents
        CONTENT_TYPES.put(".pdf", "application/pdf");
        CONTENT_TYPES.put(".doc", "application/msword");
        CONTENT_TYPES.put(".docx", "application/vnd.openxmlformats-officedocument.wordprocessingml.document");
        CONTENT_TYPES.put(".xls", "application/vnd.ms-excel");
        CONTENT_TYPES.put(".xlsx", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
        CONTENT_TYPES.put(".ppt", "application/vnd.ms-powerpoint");
        CONTENT_TYPES.put(".pptx", "application/vnd.openxmlformats-officedocument.presentationml.presentation");
        CONTENT_TYPES.put(".txt", "text/plain");
        // Video
        CONTENT_TYPES.put(".mp4", "video/mp4");
        CONTENT_TYPES.put(".3gp", "video/3gpp");
        // Audio
        CONTENT_TYPES.put(".aac", "audio/aac");
        CONTENT_TYPES.put(".mp3", "audio/mpeg");
        CONTENT_TYPES.put(".ogg", "audio/ogg");
        CONTENT_TYPES.put(".amr", "audio/amr");
        CONTENT_TYPES.put(".opus", "audio/opus");
    }

    public MediaAttachments() {
        // Load configuration from environment or .env file
        loadEnvFile();

        this.apiToken = getEnv("SENDSEVEN_API_TOKEN", null);
        this.tenantId = getEnv("SENDSEVEN_TENANT_ID", null);
        this.apiUrl = getEnv("SENDSEVEN_API_URL", "https://api.sendseven.com/api/v1");
        this.conversationId = getEnv("CONVERSATION_ID", null);

        this.httpClient = HttpClient.newHttpClient();
        this.objectMapper = new ObjectMapper();
    }

    private void loadEnvFile() {
        try {
            Properties props = new Properties();
            FileInputStream fis = new FileInputStream(".env");
            props.load(fis);
            fis.close();

            for (String key : props.stringPropertyNames()) {
                if (System.getenv(key) == null) {
                    System.setProperty(key, props.getProperty(key));
                }
            }
        } catch (IOException e) {
            // .env file not found, using environment variables only
        }
    }

    private String getEnv(String key, String defaultValue) {
        String value = System.getenv(key);
        if (value == null) {
            value = System.getProperty(key, defaultValue);
        }
        return value;
    }

    /**
     * Get content type from file extension.
     */
    public static String getContentType(String filePath) {
        String ext = "";
        int lastDot = filePath.lastIndexOf('.');
        if (lastDot > 0) {
            ext = filePath.substring(lastDot).toLowerCase();
        }
        return CONTENT_TYPES.getOrDefault(ext, "application/octet-stream");
    }

    /**
     * Get message type from content type.
     */
    public static String getMessageType(String contentType) throws Exception {
        for (Map.Entry<String, List<String>> entry : SUPPORTED_TYPES.entrySet()) {
            if (entry.getValue().contains(contentType)) {
                return entry.getKey();
            }
        }
        throw new Exception("Unsupported content type: " + contentType);
    }

    /**
     * Get maximum file size for a message type.
     */
    public static long getMaxSize(String messageType) {
        switch (messageType) {
            case "image": return IMAGE_MAX_SIZE;
            case "document": return DOCUMENT_MAX_SIZE;
            case "video": return VIDEO_MAX_SIZE;
            case "audio": return AUDIO_MAX_SIZE;
            default: return DOCUMENT_MAX_SIZE;
        }
    }

    /**
     * Upload a file as an attachment.
     */
    public JsonNode uploadAttachment(String filePath) throws Exception {
        Path path = Path.of(filePath);

        // Validate file exists
        if (!Files.exists(path)) {
            throw new FileNotFoundException("File not found: " + filePath);
        }

        long fileSize = Files.size(path);
        String filename = path.getFileName().toString();
        String contentType = getContentType(filePath);
        String messageType = getMessageType(contentType);

        // Check file size
        long maxSize = getMaxSize(messageType);
        if (fileSize > maxSize) {
            throw new Exception("File too large: " + fileSize + " bytes (max " + maxSize + " bytes for " + messageType + ")");
        }

        // Read file content
        byte[] fileContent = Files.readAllBytes(path);

        // Create multipart form data
        String boundary = "----FormBoundary" + UUID.randomUUID().toString().replace("-", "");

        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        baos.write(("--" + boundary + "\r\n").getBytes(StandardCharsets.UTF_8));
        baos.write(("Content-Disposition: form-data; name=\"file\"; filename=\"" + filename + "\"\r\n").getBytes(StandardCharsets.UTF_8));
        baos.write(("Content-Type: " + contentType + "\r\n\r\n").getBytes(StandardCharsets.UTF_8));
        baos.write(fileContent);
        baos.write(("\r\n--" + boundary + "--\r\n").getBytes(StandardCharsets.UTF_8));

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(apiUrl + "/attachments"))
                .header("Authorization", "Bearer " + apiToken)
                .header("X-Tenant-ID", tenantId)
                .header("Content-Type", "multipart/form-data; boundary=" + boundary)
                .POST(HttpRequest.BodyPublishers.ofByteArray(baos.toByteArray()))
                .build();

        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());

        if (response.statusCode() == 413) {
            throw new Exception("File too large (server rejected)");
        } else if (response.statusCode() == 415) {
            throw new Exception("Unsupported media type (server rejected)");
        } else if (response.statusCode() >= 400) {
            throw new RuntimeException("API Error " + response.statusCode() + ": " + response.body());
        }

        return objectMapper.readTree(response.body());
    }

    /**
     * Send a message with an attachment.
     */
    public JsonNode sendMediaMessage(String conversationId, String attachmentId, String messageType, String caption) throws Exception {
        ObjectNode payload = objectMapper.createObjectNode();
        payload.put("conversation_id", conversationId);
        payload.put("message_type", messageType);

        ArrayNode attachments = payload.putArray("attachments");
        attachments.add(attachmentId);

        if (caption != null && !caption.isEmpty()) {
            payload.put("text", caption);
        }

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(apiUrl + "/messages"))
                .header("Authorization", "Bearer " + apiToken)
                .header("X-Tenant-ID", tenantId)
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString(objectMapper.writeValueAsString(payload)))
                .build();

        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());

        if (response.statusCode() >= 400) {
            throw new RuntimeException("API Error " + response.statusCode() + ": " + response.body());
        }

        return objectMapper.readTree(response.body());
    }

    /**
     * Download an attachment by ID.
     */
    public void downloadAttachment(String attachmentId, String outputPath) throws Exception {
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(apiUrl + "/attachments/" + attachmentId + "/download"))
                .header("Authorization", "Bearer " + apiToken)
                .header("X-Tenant-ID", tenantId)
                .GET()
                .build();

        HttpResponse<byte[]> response = httpClient.send(request, HttpResponse.BodyHandlers.ofByteArray());

        if (response.statusCode() >= 400) {
            throw new RuntimeException("API Error " + response.statusCode() + ": " + new String(response.body()));
        }

        Files.write(Path.of(outputPath), response.body());
        System.out.println("Downloaded to: " + outputPath);
    }

    /**
     * Upload and send an image message.
     */
    public JsonNode sendImage(String conversationId, String filePath, String caption) throws Exception {
        System.out.println("Uploading image: " + filePath);
        JsonNode attachment = uploadAttachment(filePath);
        System.out.println("  Uploaded: " + attachment.get("id").asText());

        System.out.println("Sending image message...");
        return sendMediaMessage(conversationId, attachment.get("id").asText(), "image", caption);
    }

    /**
     * Upload and send a document message.
     */
    public JsonNode sendDocument(String conversationId, String filePath, String caption) throws Exception {
        System.out.println("Uploading document: " + filePath);
        JsonNode attachment = uploadAttachment(filePath);
        System.out.println("  Uploaded: " + attachment.get("id").asText());

        System.out.println("Sending document message...");
        return sendMediaMessage(conversationId, attachment.get("id").asText(), "document", caption);
    }

    /**
     * Upload and send a video message.
     */
    public JsonNode sendVideo(String conversationId, String filePath, String caption) throws Exception {
        System.out.println("Uploading video: " + filePath);
        JsonNode attachment = uploadAttachment(filePath);
        System.out.println("  Uploaded: " + attachment.get("id").asText());

        System.out.println("Sending video message...");
        return sendMediaMessage(conversationId, attachment.get("id").asText(), "video", caption);
    }

    /**
     * Upload and send an audio message.
     */
    public JsonNode sendAudio(String conversationId, String filePath, String caption) throws Exception {
        System.out.println("Uploading audio: " + filePath);
        JsonNode attachment = uploadAttachment(filePath);
        System.out.println("  Uploaded: " + attachment.get("id").asText());

        System.out.println("Sending audio message...");
        return sendMediaMessage(conversationId, attachment.get("id").asText(), "audio", caption);
    }

    /**
     * Demo: Upload a file and send it as a message.
     * Automatically detects the appropriate message type.
     */
    public void demoUploadAndSend(String filePath) throws Exception {
        String contentType = getContentType(filePath);
        String messageType = getMessageType(contentType);

        System.out.println("\n--- Sending " + messageType + " ---");
        System.out.println("File: " + filePath);
        System.out.println("Content-Type: " + contentType);

        JsonNode attachment = uploadAttachment(filePath);
        System.out.println("Attachment uploaded:");
        System.out.println("  ID: " + attachment.get("id").asText());
        System.out.println("  Filename: " + attachment.get("filename").asText());
        System.out.println("  Size: " + attachment.get("file_size").asLong() + " bytes");

        JsonNode message = sendMediaMessage(
            conversationId,
            attachment.get("id").asText(),
            messageType,
            "Here's a " + messageType + " file!"
        );

        System.out.println("Message sent:");
        System.out.println("  ID: " + message.get("id").asText());
        System.out.println("  Status: " + message.get("status").asText());
        System.out.println("  Created at: " + message.get("created_at").asText());
    }

    public void run(String[] args) {
        // Validate configuration
        if (apiToken == null || apiToken.isEmpty()) {
            System.err.println("Error: SENDSEVEN_API_TOKEN environment variable is required");
            System.exit(1);
        }

        if (tenantId == null || tenantId.isEmpty()) {
            System.err.println("Error: SENDSEVEN_TENANT_ID environment variable is required");
            System.exit(1);
        }

        if (conversationId == null || conversationId.isEmpty()) {
            System.err.println("Error: CONVERSATION_ID environment variable is required");
            System.exit(1);
        }

        System.out.println("SendSeven Media Attachments Example");
        System.out.println("========================================");
        System.out.println("API URL: " + apiUrl);
        System.out.println("Conversation: " + conversationId);

        // Check for command line argument (file to upload)
        if (args.length > 0) {
            try {
                demoUploadAndSend(args[0]);
            } catch (Exception e) {
                System.err.println("Error: " + e.getMessage());
                System.exit(1);
            }
        } else {
            System.out.println("\nUsage: java MediaAttachments <file_path>");
            System.out.println("\nSupported file types:");
            System.out.println("  Images:    .jpg, .jpeg, .png, .gif, .webp (max 16 MB)");
            System.out.println("  Documents: .pdf, .doc, .docx, .xls, .xlsx, .ppt, .pptx, .txt (max 100 MB)");
            System.out.println("  Video:     .mp4, .3gp (max 16 MB)");
            System.out.println("  Audio:     .aac, .mp3, .ogg, .amr, .opus (max 16 MB)");
            System.out.println("\nExample:");
            System.out.println("  java MediaAttachments /path/to/image.jpg");

            // Demo with a sample file if it exists
            String[] sampleFiles = {"sample.jpg", "sample.png", "sample.pdf"};
            for (String sample : sampleFiles) {
                if (new File(sample).exists()) {
                    System.out.println("\nFound sample file: " + sample);
                    try {
                        demoUploadAndSend(sample);
                    } catch (Exception e) {
                        System.err.println("Error: " + e.getMessage());
                    }
                    break;
                }
            }
        }
    }

    public static void main(String[] args) {
        new MediaAttachments().run(args);
    }
}
